src/cpuid-templates/src/lib.rs

```diff
@@ -1,1777 +1,3675 @@
 // Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use cpuid::{Cpuid, CpuidEntry, CpuidKey, CpuidRegisters, IntelCpuid, KvmCpuidFlags};
+use cpuid::{AmdCpuid, Cpuid, CpuidEntry, CpuidKey, CpuidRegisters, IntelCpuid, KvmCpuidFlags};
```

Comment:
Added a more descriptive commit body:

> Updates C3, T2 and T2S templates. Adds T2A and T2CL templates.
> Templates are formed from running `cpuid -r` within guests of the
> respective templates booted with the Firecracker main branch version on
> ec2 m5d.metal for Intel templates (C3, T2, T2S and T2CL) and ec2
> m6a.metal for AMD templates (T2A).

Reply:
How about adding "why" part that we talked offline (like: the previous one uses `__cpuid_count()` and this approach is prone to mistakes since the spec is difficult to interpret and complex. And, `cpuid` command cover things not mentioned in the official specification).

Reply:
Added the additional section:

> Previously the C3, T2 and T2S templates where generated by running a
> binary within the guest that gathered CPUID using
> `std::arch::x86_64::__cpuid_count` with an understanding of the Intel
> specification. This approach helps elide reserved leaves although
> proved to be less robust encountering multiple errors, it is difficult
> to ensure correct application of the CPUID specification. By instead
> relying upon the `cpuid` program (http://www.etallen.com/cpuid.html) we
> can ensure a robust collection of leaves even if some are unused.